edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs

```diff
@@ -0,0 +1,812 @@
+                    }
+                }
+            }
+
+            if (moduleWithDockerConfig.Config?.CreateOptions?.HostConfig?.Mounts != null)
+            {
+                foreach (DockerModels.Mount mount in moduleWithDockerConfig.Config?.CreateOptions?.HostConfig?.Mounts)
+                {
+                    if (mount.Type.Equals("bind", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        string name = KubeUtils.SanitizeDNSValue(mount.Source);
+                        string type = "DirectoryOrCreate";
+                        string hostPath = mount.Source;
+                        string mountPath = mount.Target;
+                        bool readOnly = mount.ReadOnly;
+                        volumeList.Add(new V1Volume(name, hostPath: new V1HostPathVolumeSource(hostPath, type)));
+                        volumeMountList.Add(new V1VolumeMount(mountPath, name, readOnlyProperty: readOnly));
+                    }
+                    else if (mount.Type.Equals("volume", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        string name = KubeUtils.SanitizeDNSValue(mount.Source);
+                        string mountPath = mount.Target;
+                        bool readOnly = mount.ReadOnly;
+                        volumeList.Add(new V1Volume(name, emptyDir: new V1EmptyDirVolumeSource()));
+                        volumeMountList.Add(new V1VolumeMount(mountPath, name, readOnlyProperty: readOnly));
+                    }
+                }
+            }
+
+            return volumeList.Count > 0 || volumeMountList.Count > 0
+                ? Option.Some((volumeList, proxyMountList, volumeMountList))
+                : Option.None<(List<V1Volume>, List<V1VolumeMount>, List<V1VolumeMount>)>();
+        }
+
+        private List<V1EnvVar> CollectEnv(IModule<AgentDocker.CombinedDockerConfig> moduleWithDockerConfig, KubernetesModuleIdentity identity)
+        {
+            char[] envSplit = { '=' };
+            var envList = new List<V1EnvVar>();
+            foreach (KeyValuePair<string, EnvVal> item in moduleWithDockerConfig.Env)
+            {
+                envList.Add(new V1EnvVar(item.Key, item.Value.Value));
+            }
+
+            if (moduleWithDockerConfig.Config?.CreateOptions?.Env != null)
+            {
+                foreach (string hostEnv in moduleWithDockerConfig.Config?.CreateOptions?.Env)
+                {
+                    string[] keyValue = hostEnv.Split(envSplit, 2);
+                    if (keyValue.Count() == 2)
+                    {
+                        envList.Add(new V1EnvVar(keyValue[0], keyValue[1]));
+                    }
+                }
+            }
+
+            envList.Add(new V1EnvVar(CoreConstants.IotHubHostnameVariableName, this.iotHubHostname));
+            envList.Add(new V1EnvVar(CoreConstants.EdgeletAuthSchemeVariableName, "sasToken"));
```

Comment:
> "sasToken" [](start = 81, length = 11)

While this is the only option currently, we should probably get this from whatever iotedged set this.